lesson_14/exceptions/exceptions_app/src/main/java/com/codedifferently/lesson14/ecommerce/EcommerceSystem.java

```diff
@@ -17,19 +17,29 @@ public void addProduct(String productId, String name) {
     products.put(productId, new Product(productId, name));
   }
 
-  public String placeOrder(String productId, int quantity) {
+  public String placeOrder(String productId, int quantity) throws ProductNotFoundException {
     Product product = products.get(productId);
+    if (product == null) {
+      throw new ProductNotFoundException("Product with ID 1 not found");
```

Comment:
I just realized there's a bug here. What is the `productId` isn't 1? The message should change.

Reply:
The value 1 has been changed to productId